api/v1beta1/foundationdbcluster_types.go

```diff
@@ -1205,7 +1205,7 @@ func (cluster *FoundationDBCluster) GetProcessSettings(processClass ProcessClass
 func (cluster *FoundationDBCluster) GetRoleCountsWithDefaults() RoleCounts {
 	counts := cluster.Spec.RoleCounts.DeepCopy()
 	if counts.Storage == 0 {
-		counts.Storage = 2*cluster.DesiredFaultTolerance() + 1
```

Comment:
I think we'd be better off keeping `cluster.DesiredFaultTolerance()` as a short-hand for `DesiredFaultTolerance(cluster.Spec.RedundancyMode)`, both because it's shorter and because it avoids removing a publicly exposed method.

Reply:
It doesn't make much sense to me, as DesiredFaultTolerance(cluster.Spec.RedundancyMode) needs to be public anyways. Are there any possible client calling cluster.DesiredFaultTolerance()? I mean other repos that depends of this repo.

Reply:
"Are there any possible client calling cluster.DesiredFaultTolerance()"

Since this repo is public to the world we don't know who or where this function is used and we should try our best to not break anything (even we could with the current `0.x.x` version).

Reply:
Then isn't changing RedundancyMode from string to enum gonna break things anyway?

Reply:
None of these methods are taking the redundancy mode as an argument (at least as they were). It will only break things if someone was setting the redundancy mode in the database configuration programatically. I think that latter is better than changing the method signature and it should be not that much code to have e.g.

```go
func (cluster *FoundationDBCluster) DesiredFaultTolerance() int {
  return DesiredFaultTolerance(cluster.Spec.RedundancyMode)
}
```